micro-deps/micro-deps-spring-config/src/main/java/com/ofg/infrastructure/discovery/ServiceDiscoveryInfrastructureConfiguration.java

```diff
@@ -86,18 +87,34 @@ public ServiceInstance serviceInstance(MicroserviceAddressProvider addressProvid
                 .build();
     }
 
-    @Bean(initMethod = "start", destroyMethod = "close")
+    @Bean(destroyMethod = "close")
     @SuppressWarnings("unchecked")
     public ServiceDiscovery serviceDiscovery(CuratorFramework curatorFramework,
                                              ServiceInstance serviceInstance,
                                              ServiceConfigurationResolver serviceConfigurationResolver) {
         log.info("Registering myself: " + String.valueOf(serviceInstance));
-        return ServiceDiscoveryBuilder
+        final ServiceDiscovery<Void> serviceDiscovery = ServiceDiscoveryBuilder
                 .builder(Void.class)
                 .basePath("/" + serviceConfigurationResolver.getBasePath())
                 .client(curatorFramework)
                 .thisInstance(serviceInstance)
                 .build();
+        asyncRegisterInsideZookeeper(serviceDiscovery);
+        return serviceDiscovery;
+    }
+
+    private void asyncRegisterInsideZookeeper(final ServiceDiscovery<Void> serviceDiscovery) {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    serviceDiscovery.start();
+                } catch (Exception e) {
+                    log.error("Error during service registration inside Zookeeper");
+                    System .exit(1);
```

Comment:
I like this behavior and I think it's OK for now. But would be nice to have `service.resolver.exitOnFailure` (`true` by default) option. However not essential now. This is a different feature.